drivers/disk/usdhc.c

```diff
@@ -1910,6 +1922,31 @@ static void usdhc_enable_ddr_mode(USDHC_Type *base,
 		USDHC_SYS_CTRL_SDCLKFS(prescaler);
 }
 
+static void usdhc_pinmux(struct usdhc_priv *priv, uint16_t nusdhc,
+	bool init, uint32_t speed, uint32_t strength)
+{
+#ifdef CONFIG_PINCTRL
+	/*
```

Comment:
nit: doesn't this branch require `ARG_UNUSED()` for some of the parameters? I find this function a bit confusing for the pinctrl case, callers are e.g. passing `CARD_BUS_STRENGTH_7` but it is in fact ignored. Maybe creating a separate helper would be better.

Reply:
A separate helper would work fine as well. The reason I used this approach was to limit the locations a conditional `#ifdef CONFIG_PINCTRL` was used in the code. There's only a few calls to this helper, so I'll switch the implementation to use `pinctrl_apply_state` directly there.

The function only truly needs the frequency to decide what pin settings to select (the SDK always uses a drive strength of 7 for USDHC pins), but the NXP SDK API used to configure pins accepts a drive strength, so this parameter was made available to the user.